R/baseXML.R

```diff
@@ -256,10 +256,33 @@ genBasePic <- function(imageNo) {
           <a:avLst xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main"/>
         </a:prstGeom>
       </xdr:spPr>
-    </xdr:pic>', imageNo, imageNo, imageNo)
+    </xdr:pic>', imageNo, imageNo, imageRelNo)
 }
 
-
+genLinkedPic <- function(imageNo, imageRelNo, linkRelNo) {
+  sprintf('<xdr:pic xmlns:xdr="http://schemas.openxmlformats.org/drawingml/2006/spreadsheetDrawing">
+      <xdr:nvPicPr xmlns:xdr="http://schemas.openxmlformats.org/drawingml/2006/spreadsheetDrawing">
+        <xdr:cNvPr xmlns:xdr="http://schemas.openxmlformats.org/drawingml/2006/spreadsheetDrawing" id="%s" name="Picture %s">
+          <a:hlinkClick xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships" r:id="rId%s"/>
```

Comment:
Since only this line differs between the `genBasePic()` and `genLinkedPic()`, wouldn't it be simpler to add this as another sprintf argument? Something like `ifelse(is.null(linkRelNo), "", sprintf("<a:hlinkClick...", linkRelNo))`. This would reduce duplication and make it more obvious, what is going on.

Reply:
Thanks very much for your feedback, @JanMarvin. I really appreciate it. I've now removed the unnecessary duplication, and as you say, it makes the code more simple, clear and easy to maintain.